zkabacus-crypto/src/revlock.rs

```diff
@@ -222,20 +296,91 @@ mod test {
             ]);
 
             // compare to lock using `from_bytes` method
-            assert_eq!(lock, rs.revocation_lock().0);
+            assert_eq!(lock, rp.lock.to_scalar());
         }
     }
 
     #[test]
-    pub fn checked_deserialization_works() {
+    // Test deserializtion of revocation pairs from secret
+    pub fn checked_deserialization_from_secret_works() {
         let scalar_str = "4dd70a569aa77c525dfc72b2dddd640ae1bee82b1430e63588ed71c183038d23";
         let secret =
             Scalar::from_bytes(&hex::decode(scalar_str).unwrap().try_into().unwrap()).unwrap();
 
-        let unchecked_secret = UncheckedRevocationSecret { secret, index: 0 };
-        assert!(RevocationSecret::try_from(unchecked_secret).is_err());
+        // Test that a specific invalid revocation secret produces the correct error
+        // invalid secret
+        let invalid_secret = UncheckedRevocationSecret { secret, index: 0 };
+        assert!(matches!(
+            RevocationPair::try_from(invalid_secret),
+            Err(Error::InvalidSecret())
+        ));
 
+        // Test that a specific valid revocation secret produces a verified pair
+        // valid secret
         let valid_secret = UncheckedRevocationSecret { secret, index: 1 };
-        assert!(RevocationSecret::try_from(valid_secret).is_ok())
+        assert!(RevocationPair::try_from(valid_secret).is_ok())
+    }
+
+    #[test]
+    // Check deserialization of revocation pairs from pair
+    pub fn checked_deserialization_from_pair_works() {
```

Comment:
Nit: this test doesn't explicitly test deserialization, it tests the validation functions. Can you change the name and comment? I would expect a deserialization test to have something like 
```
y = bincode::deserialize(bincode::serialize(x));
assert_eq!(y, x)
```

I would suggest writing such a test anyway (and possibly manually messing up the serialized thing) to make sure the validation functions are called correctly.

Reply:
OK, tried to address. My tests are a little repetitive now, but I'm not sure it makes sense to combine them.